ironforgedbot/common/helpers.py

```diff
@@ -16,20 +15,78 @@ def normalize_discord_string(nick: str) -> str:
     for letter in nick:
         if letter.isascii():
             new_nick.append(letter)
-    return "".join(new_nick)
+    return "".join(new_nick).strip()
 
 
-def calculate_percentage(part, whole) -> int:
-    return round(100 * float(part) / float(whole))
+def validate_user_request(
+    interaction: Interaction, playername: str
+) -> Tuple[Member, str]:
+    if not interaction.guild:
+        logging.error(f"Error accessing guild ({interaction.id})")
+        raise ReferenceError("Error accessing server")
+
+    if interaction.is_expired():
+        logging.info(f"Interaction has expired ({interaction.id})")
+        raise ReferenceError("Interaction has expired")
+
+    playername = validate_playername(playername)
+    member = find_member_by_nickname(interaction.guild, playername)
+
+    return member, playername
+
+
+def validate_playername(playername: str) -> str:
+    playername = normalize_discord_string(playername)
+
+    if len(playername) > 12 or len(playername) < 1:
+        logging.info(f"RSN length incorrect: '{playername}'")
+        raise ValueError("RSN can only be 1-12 characters long")
+
+    return playername
+
+
+def validate_protected_request(
+    interaction: Interaction, playername: str, required_role: str
+) -> Tuple[Member, str]:
+    caller, _ = validate_user_request(interaction, interaction.user.display_name)
+    member, playername = validate_user_request(interaction, playername)
 
+    has_role = validate_member_has_role(caller, required_role)
 
-def find_emoji(list: Sequence[Emoji], target: str):
-    if target in emojiCache:
-        return emojiCache[target]
+    if not has_role:
+        raise ValueError(
+            f"Member '{caller.display_name}' does not have permission for this action"
+        )
 
-    for emoji in list:
-        if emoji.available and emoji.name == target:
-            emojiCache[emoji.name] = emoji
-            return emoji
+    return member, playername
 
-    return ""
+
+def validate_member_has_role(member: Member, required_role: str) -> bool:
+    roles = member.roles
+
+    for role in roles:
+        if role.name.lower() == required_role.lower():
+            return True
+
+    return False
+
+
+def find_member_by_nickname(guild: Guild, target_name: str) -> Member:
```

Comment:
I don't think we need it, from what I can see it's used only in the `validate_user_request` and it has interaction as a param. Interaction has a member. 
If member uses command for a random name outside of discord we shouldn't prevent it

Reply:
Spoke to Kodiak about this. Leadership don't see any reason to be looking up players outside of the clan. So leaving this as is.